### ISSUE TITLE
QLP problems

### ISSUE THREAD

**Issue by vvilhelmus** [lightkurve/lightkurve#1015]
#### Problem description
A value error is generated when a QLP light curve is appended to a Kepler light curve:

#### Example

```python
import lightkurve as lk
klcc = lk.search_lightcurve('Kepler-18', cadence='long')[:18].download_all()
klc = klcc.stitch()
qlcc = lk.search_lightcurve('Kepler-18', author='QLP').download_all()
qlc = qlcc[0].append(qlcc[1])
qlc.time.format = 'bkjd'
lc1 = klc.append(qlc)
lc1.scatter()
```
The first append works as intended, but the second append gives the value error

#### Expected behavior

I expected QLP to work the same as TESS-SPOC, which does work:
```python
import lightkurve as lk
klcc = lk.search_lightcurve('Kepler-18', cadence='long')[:18].download_all()
klc = klcc.stitch()
tlcc = lk.search_lightcurve('Kepler-18', author='TESS-SPOC').download_all()
tlc = tlcc.stitch()
tlc.time.format = 'bkjd'
lc2 = klc.append(tlc)
lc2.scatter()
```
#### Environment

-  platform (e.g. Linux, OSX, Windows): Jupyter Notebook 6.1.4 in Firefox 87.0 in Linux (Pop!_OS 20.04 Ubuntu based)
-  lightkurve version (e.g. 1.0b6): 2.0.7
-  installation method (e.g. pip, conda, source): copy & pasted in notebook cell: ! python -m pip install lightkurve --upgrade

**Comment by barentsen** [comment]
Thank you for reporting this issue @vvilhelmus! :+1:

There was indeed a bug which prevented QLP and Kepler light curves from being stitched together!  This will be resolved via #1016 (which will be part of the future Lightkurve v2.0.9).  Once the fix is merged, your example will now works as follows:

<img width="693" alt="Screen Shot 2021-03-31 at 11 31 20 AM" src="https://user-images.githubusercontent.com/817669/113193294-aaf33980-9214-11eb-946c-ff9972f1e952.png">

Combining light curves from different pipelines hasn't been well tested yet, so I expect more problems to pop up.  Please do report if you encounter any!

**Comment by vvilhelmus** [comment]
Thank you so much for fixing this issue so incredible fast, Geert @barentsen ! :)
It was so fast I wasn't sure if it was an April Fools thing or not, but version 2.0.9 was downloadable and QLP light curves can be appended to Kepler light curves now. o_O Wow, nice! :D
I also see that changing the time format to sync the time-axes is redundant now as well? Something I'll have to check for myself later on. For my project I do not like the aperture used for the TESS QLP light curves, so I might be trying to make my own light curves from the TESS-cut target pixel files, see if those light curves can be combined with Kepler light curves, and do give better results. If other problems pop up from that, I will definitely report them here, again you have my gratitude! :)

**Comment by barentsen** [comment]
@vvilhelmus Sounds great!

Yes, the append/stitch functions take care of putting everything in the same time format.

It's definitely tricky to combine light curves from different pipelines.  In addition to using different aperture masks, different pipelines often apply very different post-processing steps.  For example, I believe the QLP pipeline currently applies a fairly strong filter which removes long-term trends, which will make it easier to find exoplanets but harder to find e.g. rotation period signals.  Other pipelines are tuned in different ways.

Extracting your own light curves from the pixel data will indeed be the best way to obtain light curves in a homogeneous way across many quarters/sectors, but it's definitely not an easy task, and we currently do not have a lot of experience or tutorials to demonstrate how this can best be done.  Hence, I'd be very curious to hear about what you learn, I do not thoroughly understand the best practices for doing this myself yet!

**Comment by vvilhelmus** [comment]
Hi again, Geert @barentsen 
I am currently researching the ice giant exoplanet Kepler-89e and I think the filters used in the QLP pipeline are a little too strong, because the QLP filter has removed its single transit in the TESS data:
![qlpfig](https://user-images.githubusercontent.com/81615250/113682127-1597ef80-96c3-11eb-934b-7f8f217141d9.png)
In contrast with TESS-SPOC this is kind of worrying, because when using that data it does show the single transit in the TESS data:
![spocfig](https://user-images.githubusercontent.com/81615250/113683418-77a52480-96c4-11eb-9018-3fa037c8f07d.png)
Also when I make a river plot of the combined data, it does not seem to show all of it. For the TESS observations; data seems to be removed right after phase = 0.0, when there should be something there as you can see in the above graphs.
![K89eRiverfig](https://user-images.githubusercontent.com/81615250/113684299-60b30200-96c5-11eb-91ef-ed6b50d0304c.png)
I think this is also happening for Kepler observations, where right before or after phase = 0.0 no data is shown, where I believe there should have been data shown in the river plot. There are of course gaps in river plots, but I am fairly certain they should not occur right before or after phase = 0.0.
Also I have a sneaky suspicion this is also happening in the creation of the periodogram, since the orbital period of Kepler-89e is exactly the same to the very last digit between using the Kepler-only data and using the Kepler-TESS combined data. I have no problem with the first digits being the same, but the last digits should have some deviation when additional TESS data is added.
I hope you have time to look into this.

**Comment by barentsen** [comment]
@vvilhelmus Thank you for this detailed feedback.  You raise several interesting points which need to be investigated!  Stay tuned...

**Comment by vvilhelmus** [comment]
Hello Geert @barentsen , thank you for investigating this further.
This week I have been checking other exoplanets for my MSc project and I noticed the river plots keep consistently not showing all the data, when checking different exoplanets, so this remains a problem. 
For the periodogram however, there was a difference in orbital period between Kepler-only light curves and Kepler/TESS combined light curves, when the orbital period is in the hundreds of days. So I wanted to expand on this periodogram issue a bit and this Sunday I have been experimenting with my code, since the code I have used was blindly copy&pasted from the "Identifying transiting exoplanet signals in a light curve"-tutorial without understanding what it actually means. -> http://docs.lightkurve.org/tutorials/3-science-examples/exoplanets-identifying-transiting-planet-signals.html
Let me first share the code, which generates the same orbital periods:
```python
import lightkurve as lk
import numpy as np

klcc = lk.search_lightcurve('KIC 6462863', cadence='long').download_all()
rawklc = klcc.stitch().remove_nans()
klc = rawklc.flatten(window_length=301).remove_outliers(sigma_lower=float('inf'), sigma_upper=5)

tlcc = lk.search_lightcurve('Kepler-89', author='TESS-SPOC').download_all()
rawtlc = tlcc.stitch().remove_nans()
tlc = rawtlc.flatten(window_length=301).remove_outliers(sigma_lower=float('inf'), sigma_upper=5)
lc = klc.append(tlc)

prd = np.linspace(45, 55, 10000)
kbls1 = klc.to_periodogram(method='bls', period=prd, frequency_factor=200)
bls1 = lc.to_periodogram(method='bls', period=prd, frequency_factor=200)
print(kbls1.period_at_max_power, bls1.period_at_max_power)
```
So there is some NumPy code in there, which has 3 values as "linspace" where I understand the first two values represent the minimum and the maximum period, but I have no idea what the third value represents. I have changed the third value from ten thousand to one hundred thousand and that seems to make the calculation longer and also more accurate, but the orbital periods it produces are exactly the same (unless the orbital period of the exoplanet is in the hundreds of days).
Also I see that third value return in the description column of the lightkurve.Periodogram properties as: array (100000,) , if I have used one hundred thousand as the third value, when I run the Python code:
```python
kbls1.show_properties()
```
or
```python
bls1.show_properties()
```
And when I avoid using the NumPy coding by using the following LightKurve-only code:
```python
kbls2 = klc.to_periodogram(method='bls', minimum_period=45, maximum_period=55, frequency_factor=10)
bls2 = lc.to_periodogram(method='bls', minimum_period=45, maximum_period=55, frequency_factor=10)
print(kbls2.period_at_max_power, bls2.period_at_max_power)
```
The two orbital periods are no longer the same, which is a good thing. However, when I show the properties with:
```python
kbls2.show_properties()
```
and
```python
bls2.show_properties()
```
the values in the arrays are no longer the same either. Also when I reduce the frequency factor value from 10 to 1, the array values increase and the orbital periods seem to be getting more accurate along with a much longer calculation.
So my question is; what is good practice when comparing the orbital periods obtained from a Kepler-only light curve and a Kepler/TESS combined light curve? Would similar Array values be required? And what does that third NumPy value/Array value actually mean?
Sorry, I am still a beginner and still learning things, but I hope someone can teach me something here. :confused:
